### PR TITLE
Update pinmux mapping diagram to match `top_config.toml`

### PR DESCRIPTION
**Pull request by AlexJones0**
See the commit messages for more details. This PR only updates the pin mapping diagram to fix issues in the diagram and to make it match the latest `top_config.toml` configuration, and makes no change to any code (SW/HW) at all - just the documentation.

The changes include:
  - Swapping RPH SPI CE0&1 - this is already correct in `top_config.toml`, but the diagram gets these two chip enables the wrong way around. See [`top_config.toml`](https://github.com/lowRISC/sonata-system/blob/9f794fe3bd4eec8d1a01ee81da97a7f2cec0b452/data/top_config.toml#L155-L167) and [pinout.xyz](https://pinout.xyz/pinout/pin26_gpio7/) to see why this change is necessary.
  - Updates the SPI names, so that `spi_board` now becomes `spi_0`, `spi_0` becomes `spi_1` and `spi_1` becomes `spi_2`, to represent the corresponding changes to `top_config.toml` after the flash Flash/MicroSD SPI was added back through pinmux. See #326 where this change was made.
  - Changes the `PMOD{0,1}` pin names to match the corresponding names in `top_config.toml`. That is, it changes names like `PMOD0_5`, `PMOD0_6`, `PMOD0_7` and `PMOD0_8` to `PMOD0_7`, `PMOD0_8`, `PMOD0_9` and `PMOD0_10` to better match the PMOD names used in the Digilent specification. Again see #326 where this change was made.
  - Fixes a small typo for `PMOD1_4`: "sca" -> "sda". See the error [here](https://github.com/lowRISC/sonata-system/blob/0ec0807fb740d8f14dd8a7cc2d8097bc56cbd1e0/doc/ip/pinmux/pin_mappings.dot#L308) and the correct line in `top_config.toml` [here](https://github.com/lowRISC/sonata-system/blob/9f794fe3bd4eec8d1a01ee81da97a7f2cec0b452/data/top_config.toml#L504).
  - Adds the RS485 RX and TX pins to the pin mapping diagram. Designed to match the description of RS485 given in [`top_config.toml`](https://github.com/lowRISC/sonata-system/blob/9f794fe3bd4eec8d1a01ee81da97a7f2cec0b452/data/top_config.toml#L88-L95).
 - Updates the Raspberry Pi HAT header's pin names to remove references to SPI pins. See commit e0e3a2339f133c44f9c8937b538a7cdaaeeb79d5 of #326 again to see where this change was made.

### CODE REVIEW

**Review by marnovandermaas — DISMISSED** [review]
Yup, that's the right way around.

**Comment by AlexJones0** [comment]
Latest force push addresses other inconsistencies in the pin mapping diagram: it updates the SPI names/numbering, fixes a typo, changes the PMOD0 and PMOD1 pin name numbering to match `top_config.toml`, and adds the missing RS485 pins.

File: doc/ip/pinmux/pin_mappings.dot [hunk 1]
```diff
@@ -485,8 +496,8 @@ digraph {
   gpio_rph:"gpio_0-27" -> rph:g4;
   gpio_rph:"gpio_0-27" -> rph:g5;
   gpio_rph:"gpio_0-27" -> rph:g6;
-  gpio_rph:"gpio_0-27" -> rph:g7_spi0_ce0;
-  gpio_rph:"gpio_0-27" -> rph:g8_spi0_ce1;
+  gpio_rph:"gpio_0-27" -> rph:g7_spi0_ce1;
+  gpio_rph:"gpio_0-27" -> rph:g8_spi0_ce0;
```
Comment (HU90m) [inline doc/ip/pinmux/pin_mappings.dot:500]: Would you be able to remove the SPI suffixes from the Raspberry Pi Header pins?

```suggestion
  gpio_rph:"gpio_0-27" -> rph:g7;
  gpio_rph:"gpio_0-27" -> rph:g8;
```

**Review by HU90m — APPROVED** [review]
great! thanks for doing this Alex

File: doc/ip/pinmux/pin_mappings.dot (hunk 1 above)
Comment (AlexJones0) [reply doc/ip/pinmux/pin_mappings.dot]: Added in the latest commit.

**Review by marnovandermaas — APPROVED** [review]

**Pull request merged**